src/Message.php

```diff
@@ -16,7 +16,7 @@
     public function __construct($payload, $timestamp, $encoding)
     {
         $this->payload = $payload;
-        $this->timestamp = new \DateTimeImmutable($timestamp);
+        $this->timestamp = new \DateTimeImmutable("@$timestamp");
```

Comment:
We need to convert before passing to Message object then...
Maybe we need to receive in the construct the *\DateTimeImmutable* instead of a timestamp....